tests/integration/long/test_failure_types.py

```diff
@@ -0,0 +1,295 @@
+import time
+from tests.integration import use_singledc, PROTOCOL_VERSION, get_cluster, setup_keyspace
+
+
+def setup_module():
+    """
+    We need some custom setup for this module. All unit tests in this module
+    require protocol >=4. We won't bother going through the setup required unless that is the
+    protocol version we are using.
+    """
+
+    # If we aren't at protocol v 4 or greater don't waste time setting anything up, all tests will be skipped
+    if PROTOCOL_VERSION >= 4:
+        use_singledc(start=False)
+        ccm_cluster = get_cluster()
+        ccm_cluster.stop()
+        config_options = {'tombstone_failure_threshold': 2000, 'tombstone_warn_threshold': 1000}
+        ccm_cluster.set_configuration_options(config_options)
+        ccm_cluster.start(wait_for_binary_proto=True, wait_other_notice=True)
+        # there seems to be some race, with some versions of C* taking longer to
+        # get the auth (and default user) setup. Sleep here to give it a chance
+        time.sleep(10)
+        setup_keyspace()
+
+
+def teardown_module():
+    """
+    The rests of the tests don't need custom tombstones
+    reset the config options so as to not mess with other tests
+    """
+    if PROTOCOL_VERSION >= 4:
+        ccm_cluster = get_cluster()
+        config_options = {}
+        ccm_cluster.set_configuration_options(config_options)
+        if ccm_cluster is not None:
+            ccm_cluster.stop()
+
+
+class ClientExceptionTests(unittest.TestCase):
+
+    def setUp(self):
+        """
+        Test is skipped if run with native protocol version <4
+        """
+
+        if PROTOCOL_VERSION < 4:
+            raise unittest.SkipTest(
+                "Native protocol 4,0+ is required for custom payloads, currently using %r"
+                % (PROTOCOL_VERSION,))
+
+        self.cluster = Cluster(protocol_version=PROTOCOL_VERSION)
+        self.session = self.cluster.connect()
+        self.nodes_currently_failing = []
+        self.node1, self.node2, self.node3 = get_cluster().nodes.values()
+
+    def tearDown(self):
+
+        self.cluster.shutdown()
+        failing_nodes = []
+
+        # Restart the nodes to fully functional again
+        self.setFailingNodes(failing_nodes, "testksfail")
+
+    def setFailingNodes(self, failing_nodes, keyspace):
+        """
+        This method will take in a set of failing nodes, and toggle all of the nodes provided to fail writes. It will
+        @param failing_nodes A definitive list of nodes that should fail writes
+        @param keyspace The keyspace to enable failures on
+
+        """
+
+        # Ensure all of the nodes on the list have failures enabled
+        for node in failing_nodes:
+            if node not in self.nodes_currently_failing:
+                print "Stopping node "+node.name
+                node.stop(wait_other_notice=True, gently=False)
+                print "starting crippled node "+node.name
```

Comment:
pep8 whitespace. Can probably change to formatting if you choose to use logging for these messages.